src/main.py

```diff
@@ -104,13 +98,27 @@ def render_page(sensors):
     measure = 'GET /?measure=on' in request
 
     # note that LED is on on low signal
+    measurements = []
     if measure:
         for sensor in sensors:
-            sensor.measure()
+            pass
+            measurements.append(sensor.measure())  # yields a future
+
+    # wait for measurements to actually finish
+    await uasyncio.gather(*measurements)
 
     response = render_page(sensors)
-    conn.send('HTTP/1.1 200 OK\n')
-    conn.send('Content-Type: text/html\n')
-    conn.send('Connection: close\n\n')
-    conn.sendall(response)
-    conn.close()
+    writer.write('HTTP/1.1 200 OK\n')
+    writer.write('Content-Type: text/html\n')
+    writer.write(response)
+    await writer.drain()
+    await writer.wait_closed()
+
+
+loop = uasyncio.get_event_loop()
+loop.create_task(uasyncio.start_server(serve, "0.0.0.0", 80))
```

Comment:
`uasyncio` provides a convenient method to start a tcp server. It needs a single callback for any incoming requests. We ask it to bind to any connection (`0.0.0.0`) on port 80.